client/components/form/Form.jsx

```diff
@@ -53,22 +54,26 @@ export default function Form({ email }) {
     dayjs().hour(10).minute(0).second(0)
   );
   const [address, setAddress] = useState("");
+  const [disableBtn, setDisableBtn] = useState(null);
+  // Error state
   const [errorMsg, setErrorMsg] = useState("");
   const [errorPath, setErrorPath] = useState("");
   const [toast, setToast] = useState(false);
   const [toastMsg, setToastMsg] = useState("");
-  const [disableBtn, setDisableBtn] = useState(null);
+  const [hasSubmitError, setHasSubmitError] = useState(false);
   const [isPending, setIsPending] = useState(false);
 
   function handleCancel() {
-    redirect("/new-appointment/cancel");
+    router.push("/new-appointment/cancel");
```

Comment:
In case anyone else is curious why this change, this [comment](https://old.reddit.com/r/nextjs/comments/17gsis3/when_would_i_use_redirect_vs_routerpush_in_next_13/k6imjwz/) seems to explain

> Redirect and router.push() are fairly similar, the main differences are: - router.push() can only be used on the client - redirect can be used on the client and server - only router.push() pushes a new entry into the history stack so on back navigation the user will return to where they previously were
> 
> Typically unless you’re on the server or do not want to push a new entry to the history stack you should always use router.push()

Reply:
Yes, correct. Although `redirect` will work in a client component, it will not work in a try/catch, and when its placed outside the try/catch, it will still execute even if an error has occurred. `router.push()` is the correct option for client components and try/catch.